packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,237 @@
+     *          conclusion of fast relay. This address is the "owner" of the deposit box.
+     * @param _minimumBridgingDelay Minimum second that must elapse between L2->L1 token transfer to prevent dos.
+     * @param timerAddress Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(
+        address _l1WithdrawContract,
+        uint64 _minimumBridgingDelay,
+        address timerAddress
+    ) OVM_CrossDomainEnabled(Lib_PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER) OVM_Testable(timerAddress) {
+        l1WithdrawContract = _l1WithdrawContract;
+        minimumBridgingDelay = _minimumBridgingDelay;
+    }
 
-    function setWithdrawalContract(address withdrawalContract) public onlyFromCrossDomainAccount(l1Owner) {}
+    /**************************************
+     *          ADMIN FUNCTIONS           *
+     **************************************/
 
-    function whitelistToken(address l1Token, address l2Token) public onlyFromCrossDomainAccount(l1Owner) {}
+    /**
+     * @notice Changes the L1 withdraw associated with this L2 deposit box.
+     * @dev Only callable by the existing l1WithdrawContract via the optimism cross domain messenger.
+     * @param newL1WithdrawContract address of the new L1 withdrawContract.
+     */
+    function changeL1WithdrawContract(address newL1WithdrawContract)
+        public
+        onlyFromCrossDomainAccount(l1WithdrawContract)
+    {
+        emit L1WithdrawContractChanged(l1WithdrawContract, newL1WithdrawContract);
+        l1WithdrawContract = newL1WithdrawContract;
+    }
 
-    function disableDepositContract() public onlyFromCrossDomainAccount(l1Owner) {}
+    /**
+     * @notice Enables L1 owner to whitelist a L1 Token <-> L2 Token pair for bridging.
+     * @dev Only callable by the existing l1WithdrawContract via the optimism cross domain messenger.
+     * @param l1Token Address of the canonical L1 token. This is the token users will receive on Ethereum.
+     * @param l2Token Address of the L2 token representation. This is the token users would deposit on optimism.
+     */
+    function whitelistToken(address l1Token, address l2Token) public onlyFromCrossDomainAccount(l1WithdrawContract) {
+        whitelistedTokens[l2Token] = WhitelistedToken({ l1Token: l1Token, lastBridgeTime: uint64(getCurrentTime()) });
 
-    function transferL1Owner(address _l1Owner) public onlyFromCrossDomainAccount(l1Owner) {
-        l1Owner = _l1Owner;
+        emit TokenWhitelisted(l1Token, l2Token);
     }
 
-    // Depositor functions
+    /**
+     * @notice L1 owner can enable/disable deposits over all whitelisted tokens.
+     * @dev Only callable by the existing l1WithdrawContract via the optimism cross domain messenger.
+     * @param _depositsEnabled bool to set if the deposit box should accept/reject deposits.
+     */
+    function setEnableDeposits(bool _depositsEnabled) public onlyFromCrossDomainAccount(l1WithdrawContract) {
```

Comment:
Thoughts on renaming this to `resetEnableDeposits` or `toggleEnableDeposits` since there are no practical cases where you'd set `depositsEnabled = true` if its already `true` (same with `false`). You'd only want to flip the switch.

Reply:
Ok SGTM